actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -45,6 +50,14 @@ def listener
           @listener || @server.mutex.synchronize { @listener ||= Listener.new(self, @server.event_loop) }
         end
 
+        def verify_and_yield!(pg_conn, &block)
```

Comment:
Think this handles two separate concerns: verifying and satisfying the surrounding code's usage.

`_and_yield!` also reads like a smell. Ruby code usually doesn't advertise its yieldyness 😊

I think moving yield out to the callers and going with this would be clearer:

```ruby
def verify(conn)
  unless conn.is_a?…
    raise …
  end
end
```